newIDE/app/src/MainFrame/index.js

```diff
@@ -2083,19 +2093,23 @@ const MainFrame = (props: Props) => {
     ]
   );
 
+  /**
+   * Returns true if the project has been closed and false if the user refused to close it.
+   */
   const askToCloseProject = React.useCallback(
-    (): Promise<void> => {
-      if (unsavedChanges && unsavedChanges.hasUnsavedChanges) {
-        if (!currentProject) return Promise.resolve();
+    async (): Promise<boolean> => {
+      if (!currentProject) return true;
 
+      if (unsavedChanges && unsavedChanges.hasUnsavedChanges) {
         const answer = Window.showConfirmDialog(
```

Comment:
maybe not for this PR, but we should take some time to migrate all those showConfirmDialog to the new `useConfirmDialog`

Reply:
Yep, true!